contexts/Aladin.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+import {
+  createContext,
+  FunctionComponent,
+  MutableRefObject,
+  PropsWithChildren,
+  RefCallback,
+  useCallback,
+  useContext,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  Aladin,
+  AladinInstance,
+  AladinOptions,
+  ImageHiPSOptions,
+} from "@/types/aladin";
+import defaultOptions, {
+  defaultHiPSOptions,
+} from "@/fixtures/defaultAladinOptions";
+import { useOnClickOutside } from "@/hooks/listeners";
+
+interface AladinProps {
+  fovRange?: Array<number>;
+  options?: AladinOptions;
+  hipsConfig: {
+    id: string;
+    options: ImageHiPSOptions;
+  };
+}
+
+interface AladinContextValue {
+  aladin: AladinInstance | null;
+  A: Aladin | null;
+  setRef: RefCallback<HTMLElement>;
+  ref: MutableRefObject<HTMLElement | null>;
+  hasFocus: boolean;
+}
+
+const AladinContext = createContext<AladinContextValue | null>(null);
+
+export const AladinProvider: FunctionComponent<
+  PropsWithChildren<AladinProps>
+> = ({ children, fovRange = [2, 90], hipsConfig, options = {} }) => {
+  const [aladin, setAladin] = useState<AladinInstance | null>(null);
+  const [A, setGlobalInstance] = useState<Aladin | null>(null);
+
+  const [hasFocus, setFocus] = useState(false);
+
+  const renderRef = useRef<HTMLElement | null>(null);
+  const setRef: RefCallback<HTMLElement> = useCallback((node: HTMLElement) => {
+    renderRef.current = node;
+  }, []);
+
+  const onFocus = () => {
+    setFocus(true);
+  };
+
+  const onBlur = () => {
+    setFocus(false);
+  };
+
+  useLayoutEffect(() => {
```

Comment:
this layout effect should run early in the page render and load the library then hold it in React's state, once it init's an instance of `aladin` is created with the provided config and sets the survey to the provided HiPS catalog